pages/base.py

```diff
@@ -15,89 +17,200 @@ def __init__(self, driver):
+                timeout = self.locators[element][2]
+            location = (self.locators[element][0], self.locators[element][1])
 
             try:
-                element = WebDriverWait(self.driver, timeout).until(
+                WebDriverWait(self.driver, timeout).until(
                     EC.presence_of_element_located(location)
                 )
-            except(TimeoutException,StaleElementReferenceException):
-                raise ValueError('Element {} not present on page'.format(what))
+            except(TimeoutException, StaleElementReferenceException):
+                raise ValueError('Element {} not present on page. {}'.format(element, self.driver.current_url))
 
             try:
-                element = WebDriverWait(self.driver, timeout).until(
+                WebDriverWait(self.driver, timeout).until(
                     EC.visibility_of_element_located(location)
                 )
-            except(TimeoutException,StaleElementReferenceException):
-                raise ValueError('Element {} not visible before timeout'.format(what))
-            # I could have returned element, however because of lazy loading, I am seeking the element before return
+            except(TimeoutException, StaleElementReferenceException):
+                raise ValueError('Element {} not visible before timeout. {}'.format(element, self.driver.current_url))
+
+            if 'link' in element:
+                try:
+                    WebDriverWait(self.driver, timeout).until(
+                        EC.element_to_be_clickable(location)
+                    )
+                except(TimeoutException, StaleElementReferenceException):
+                    raise ValueError('Element {} on page but not clickable. {}'.format(element, self.driver.current_url))
+
             return self.find_element(*location)
         else:
-            raise ValueError('Cannot find element {}'.format(what))
+            raise ValueError('Cannot find element {}. {}'.format(element, self.driver.current_url))
+
 
 class BasePage(BaseElement):
     url = None
 
-    def open(self):
+    def goto(self):
         self.driver.get(self.url)
+        self.check_page()
+
+    def check_page(self):
+        if not self.verify():
+            # handle any specific kind of error before go to page exception
+            self.error_handling()
+            raise PageException('Unexpected page structure: `{}` with identity {}'.format(
+                self.driver.current_url, self.locators['identity']
+            ))
+
+    def verify(self):
+        try:
+            self.identity
+        except ValueError:
+            return False
+        else:
+            return True
+
+    def error_handling(self):
+        pass
+
+    def reload(self):
+        self.driver.refresh()
+
+
+class Navbar(BaseElement):
+
+    locators = {
+        'service_dropdown': (By.CSS_SELECTOR, '#navbarScope > div.container > div.navbar-header > div.dropdown > button'),
```

Comment:
I'd rather have these selecting by IDs sometime in the near future. It's fine for now, but ID is much better than selectors which are faaar better than XPATHs.